### ISSUE TITLE
Microsoft.Build.Utilities version 2.0.0.0 still referenced

### ISSUE THREAD

**Issue by ma-bjo** [wixtoolset/issues#5711]
* Which version of WiX are you building with?

> v3.14.0.712 and v4.0.0.5512

* Which version of Visual Studio are you building with (if any)?

> MSBuild 15 from Visual Studio Professional 2017

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> None.

* Which version of .NET are you building with?

> 4.7

* Describe the problem and the steps to reproduce it.

> The old assembly Microsoft.Build.Utilities, Version=2.0.0.0 is not available in a build environment where .NET Framework 3.5 has not been installed. Up until version 3.11 this is fine, but for WiX 3.14 and 4.0 this assembly should not be required. However, looking at the manifest for WixTasks.dll with ildasm.exe I can see that it still references this legacy assembly.

> I am fully aware of that none of these versions are released yet, but I though I should let you know that this needs to be fixed.

* Describe the behavior you expected and how it differed from the actual behavior.

> .NET Framework 3.5 should not be required anymore.

**Comment by clisc** [comment]
This is an issue for me as well.

**Comment by tesqie** [comment]
Still an Issue for me as well.

**Comment by maxnelso** [comment]
Any updates on this?

**Comment by DamonStamper** [comment]
Reporting similar behavior.

**Comment by BMurri** [comment]
I would like to point out that the issue is still open, so I wouldn't expect it to be resolved yet.

The workaround continues to be installing the .NET 3.5 framework (which can still be done on all supported Windows versions/editions).

**Comment by clisc** [comment]
Hi Blair,

Yes, I understand the workaround.  However, we are trying to remove all dependencies .NET 2.0 and .NET 3.5 from our build system so that new installs of Windows Server 2016 work out of the box.

It would be nice to have a version compiled against a 4.x version of the .NET framework so the down-level .NET framework dependencies can be removed.

Thanks,
Cliff

**Comment by BMurri** [comment]
I actually agree this should be resolved. The resources involved have been tied up with other priorities, and now additional resources would speed things up. If you can volunteer resources, or fund resources, it would speed up resolving this issue.

**Comment by robmen** [comment]
@clisc as @BMurri points out it's a matter of resources. This issue will be resolved but WiX v4 is our priority and WiX v3.14 (where this fix will go) quickly follows up WiX v4 to help migrate form v3.x to v4.

**Comment by robmen** [comment]
My guess right now is early to mid-next year. Currently, the bulk of the development is being done by @firegiantco employees. How much time they have to dedicate to WiX v4 is very dependent on the needs and desires of their customers.

Want to enhance WiX v4 progress? I see two options: 1) jump in and contribute features required to release WiX v4 directly or 2) become a @firegiantco customer and help them make WiX v4 a priority.

**Comment by robmen** [comment]
@ma-bjo those WiX v4 builds are remnants from when WiX v3 and WiX v4 builds were kicked off in parallel. It does not represent WiX v4 at all. The current architecture of WiX v4 is better discussed in this [WiX Online Meeting](https://www.firegiant.com/blog/2018/10/25/wix-online-meeting-157-highlights/).

**Comment by jozefizso** [comment]
This was fixed in Dec 2018 in the wix3 source code, but the wix v3.11.2 (released on Sep 2019) still ships with **WixTasks.dll** which requires the old **Microsoft.Build.Utilities, Version=2.0.0.0** assembly.

When can we expect the fix to be in official release?

**Comment by barnson** [comment]
It's in v3.14.

**Comment by ma-bjo** [comment]
@barnson But that version is not a stable release, right? On https://wixtoolset.org/ the latest production build is said to be 3.11.2. Version 3.14 is only available under development builds and that page specifically states that "These releases may be unstable and are therefore not recommended for production use."

Is that statement true, or is it just the web page that needs to be updated with new information?

**Comment by BMurri** [comment]
That statement is still true.

**Comment by DaviesCooper** [comment]
My assumption is that I am doing something wrong, but I keep getting this same error using wix 3.11.1 and .net framework 4.6.1. Does this also pertain to .net 4.6.1?

**Comment by BMurri** [comment]
It's "fixed" in v3.14, which hasn't been officially released yet. I would expect WiX to still require .net v3x for all WiX v3 versions before v3.14.
